nanodbc/nanodbc.cpp

```diff
@@ -3161,6 +3165,37 @@ inline void result::result_impl::get_ref_impl<std::vector<std::uint8_t>>(
     throw type_incompatible_error();
 }
 
+template <>
+void result::result_impl::get_ref_from_string_column(short column, string::value_type& result) const
+{
+    bound_column& col = bound_columns_[column];
+    const char* s = col.pdata_ + rowset_position_ * col.clen_;
+    switch (col.ctype_)
+    {
+    case SQL_C_CHAR:
+        result = (string::value_type) * (const char*)(s);
```

Comment:
Yah, we try to avoid C-style casts (even tho C++ casts are indeed long and ugly) because they're more obvious and generally safer (can't accidentally cast away const-ness, for example).